artifactory_cleanup/rules/delete.py

```diff
@@ -92,37 +93,85 @@ class delete_empty_folder(Rule):
     """
 
     def _aql_add_filter(self, aql_query_list):
-        update_dict = {
-            "repo": {
-                "$match": "deleteEmptyFolder",
-            }
-        }
-        aql_query_list.append(update_dict)
+        # Get list of all files and folders
```

Comment:
I remember that we used the plugin because it works much faster then our python code. 
How much folders\artifacts do you have in your repo? 

Especiialy because of this:
```python
        # Get list of all files and folders
        all_files_dict = {"path": {"$match": "**"}, "type": {"$eq": "any"}}
```

If you can test it somehow with a repository with ~50K-100K artifacts in it it'd be great!

Reply:
I tested on various repositories on our setup, with around 20k artifacts.
There it worked quite fast (i.e., took around 5-10 seconds to download the file).

Seems good enough.  Another optimization would be to not return all properties, but only `name` and `path`. But did not want to change this in the code for now